source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -377,6 +377,40 @@ absl::optional<SystemTime> SslSocket::expirationPeerCertificate() const {
   return Utility::getExpirationTime(*cert);
 }
 
+bool SslSocket::peerCertificateChainPresented() const {
+  bssl::UniquePtr<STACK_OF(X509)> cert_chain(SSL_get_peer_cert_chain(ssl_.get()));
+  return cert_chain != nullptr && sk_X509_num(cert_chain.get()) > 0;
```

Comment:
this is very similar to `peerCertificatePresented()`, what is the case that one is true and other is false?

Reply:
I'm unsure if this can happen in reality, but I've added it as an additional safety check and for consistency.

Reply:
@PiotrSikora?